examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect.ino

```diff
@@ -0,0 +1,591 @@
+    uint8_t* encodedSignedSignature, size_t encodedSignedSignatureSize,
+    size_t* encodedSignedSignatureLength);
+static uint64_t getSASTokenExpirationTime(uint32_t minutes);
+
+// Time and Error functions.
+static unsigned long getTime();
+static String getFormattedDateTime(unsigned long epochTimeInSeconds);
+static String mqttErrorCodeName(int errorCode);
+
+/*---------------------------*/
+/*    Main code execution    */
+/*---------------------------*/
+
+/*
+ * setup:
+ * Initialization and connection of serial communication, WiFi client, Azure IoT SDK for C client, 
+ * and MQTT client.
+ */
+void setup() 
+{
+  while (!Serial);
+  Serial.begin(MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE);
+  pinMode(LED_PIN, OUTPUT);
+
+  digitalWrite(LED_PIN, HIGH);
+
+  connectToWiFi();
+  initializeAzureIoTHubClient();
+  initializeMQTTClient();
+  connectMQTTClientToAzureIoTHub();
+
+  digitalWrite(LED_PIN, LOW);
+}
+
+/*
+ * loop:
+ * Check for connection and reconnect if necessary.
+ * Send Telemetry and receive messages.
+ */
+void loop() 
+{
+  if (WiFi.status() != WL_CONNECTED) 
+  {
+    connectToWiFi();
+  }
+
+  // Telemetry
+  if (millis() > telemetryNextSendTimeMs) 
+  {
+    // Check for MQTT Client connection to Azure IoT hub. Reconnect if needed.
+    if (!mqttClient.connected()) 
+    {
+      connectMQTTClientToAzureIoTHub();
+    }
+
+    sendTelemetry();
+    telemetryNextSendTimeMs = millis() + IOT_CONFIG_TELEMETRY_FREQUENCY_MS;
+  }
+
+  // MQTT loop must be called to process Telemetry and Cloud-to-Device (C2D) messages.
+  mqttClient.poll();
+  delay(500);
+}
+
+/*-----------------------------------------------*/
+/*    Initialization and connection functions    */
+/*-----------------------------------------------*/
+
+/*
+ * connectToWifi:
+ * The WiFi client connects, using the provided SSID and password.
+ * The WiFi client synchronizes the time on the device. 
+ */
+void connectToWiFi() 
+{
+  logString = "Attempting to connect to WIFI SSID: ";
+  LogInfo(logString + IOT_CONFIG_WIFI_SSID);
+
+  while (WiFi.begin(IOT_CONFIG_WIFI_SSID, IOT_CONFIG_WIFI_PASSWORD) != WL_CONNECTED) 
+  {
+    Serial.print(".");
+    delay(IOT_CONFIG_WIFI_CONNECT_RETRY_MS);
+  }
+  Serial.println();
+
+  logString = "WiFi connected, IP address: ";
+  LogInfo(logString + WiFi.localIP() + ", Strength (dBm): " + WiFi.RSSI());
+  LogInfo("Syncing time.");
+
+  while (getTime() == 0) 
+  {
+    Serial.print(".");
+    delay(500);
+  }
+  Serial.println();
+
+  LogInfo("Time synced!");
+}
+
+/*
+ * initializeAzureIoTHubClient:
+ * The Azure IoT SDK for C client uses the provided hostname, device id, and user agent.
+ */
+void initializeAzureIoTHubClient() 
+{
+  LogInfo("Initializing Azure IoT Hub client.");
+
+  az_span hostname = AZ_SPAN_FROM_STR(IOT_CONFIG_IOTHUB_FQDN);
+  az_span deviceId = AZ_SPAN_FROM_STR(IOT_CONFIG_DEVICE_ID);
+
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();

```

Comment:
Note for the Azure IoT C SDK Team: options here is done the same way as in the other existing samples, and its scoped to this function, initializeAzureIoTHubClient. Once the function exits, `options` looses its scope. However, it's permanently referenced in `azIoTHubClient`, which is global. This could be a potential issue for customers reusing the code.